moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -267,19 +270,32 @@ bool PlanningPipeline::generatePlan(const planning_scene::PlanningSceneConstPtr&
   // ---------------------------------
   // Solve the motion planning problem
   // ---------------------------------
-
   planning_interface::MotionPlanRequest mutable_request = req;
   try
   {
+    using clock = std::chrono::system_clock;
+    const auto plan_start_time = clock::now();
+    const double allowed_planning_time = req.allowed_planning_time;
+
     // Call plan request adapter chain
     for (const auto& req_adapter : planning_request_adapter_vector_)
     {
       assert(req_adapter);
       RCLCPP_INFO(node_->get_logger(), "Calling PlanningRequestAdapter '%s'", req_adapter->getDescription().c_str());
       const auto status = req_adapter->adapt(planning_scene, mutable_request);
       res.error_code = status.val;
+      std::string message = status.message;
+
       // Publish progress
       publishPipelineState(mutable_request, res, req_adapter->getDescription());
+
+      // check for timeout (mainly for sanity check since adapters are fast)
+      if (std::chrono::duration<double>(clock::now() - plan_start_time).count() >= allowed_planning_time)
+      {
+        message = "failed to finish within " + std::to_string(allowed_planning_time) + "s";
```

Comment:
```suggestion
        status.message = "Failed to finish planning within allowed planning time ( " + std::to_string(allowed_planning_time) + "s )";
```

Reply:
No need to write short error messages. Be as precise and long as necessary :wink: (If you accept this change you'll still need to format the code correctly with pre-commit)